portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -27,30 +29,48 @@
 /** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  private List<String> comments = new ArrayList<>(
-    Arrays.asList("Nice!",
-                  "Very interesting.",
-                  "I would like to know more about your favourite TV shows!"));
+  private final Deque<Comment> comments = new LinkedList<>();
 
   /**
-   * Converts a List instance into a JSON string.
+   * Converts a Deque instance into a JSON string.
    */
-  public static String convertListToJson(List<String> list) {
-    StringBuilder jsonBuilder = new StringBuilder();
-    jsonBuilder.append("[");
-    for (int i = 0; i < list.size(); i++) {
-      jsonBuilder.append("\"" + list.get(i) + "\"");
-      if (i < list.size() - 1) {
-        jsonBuilder.append(", ");
-      }
-    }
-    jsonBuilder.append("]");
-    return jsonBuilder.toString();
+  public static String convertDequeToJson(Deque<Comment> deque) {
+    Gson gson = new Gson();
+    String json = gson.toJson(deque);
+    return json;
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(MediaType.APPLICATION_JSON);
-    response.getWriter().println(convertListToJson(comments));
+    response.getWriter().println(convertDequeToJson(comments));
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  public static String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
this can probably be private

Reply:
so `convertDequeToJson` could be private too, right?

Reply:
Correct. Anything you (or the framework) don't call from outside can be private.